File: src/main/kotlin/top/charjin/shoppingserver/mapper/OsGoodsMapper.java
package top.charjin.shoppingserver.mapper;

import org.apache.ibatis.annotations.Param;
import org.jetbrains.annotations.Nullable;
import top.charjin.shoppingserver.entity.OsGoods;
import top.charjin.shoppingserver.model.OsGoodsModel;

import java.util.List;

public interface OsGoodsMapper {
    int deleteByPrimaryKey(Integer id);

    int insert(OsGoods record);

    int insertSelective(OsGoods record);

    OsGoods selectByPrimaryKey(Integer id);

    int updateByPrimaryKeySelective(OsGoods record);

    int updateByPrimaryKey(OsGoods record);

    List<OsGoodsModel> selectGoodsByShopId(@Param("userId") int userId, @Param("shopId") int shopId);


    List<OsGoods> selectAllGoods();

    List<OsGoods> getGoodsByKey(@Nullable String key);


    List<OsGoods> getAllGoodsByShopId(@Param("shopId") int shopId);
}

File: src/main/kotlin/top/charjin/shoppingserver/mapper/OsGoodsMapper.kt
package top.charjin.shoppingserver.mapper

import org.apache.ibatis.annotations.Param
import top.charjin.shoppingserver.entity.OsGoods

interface OsGoodsMapper {
    fun deleteByPrimaryKey(id: Int?): Int

    fun insert(record: OsGoods): Int

    fun insertSelective(record: OsGoods): Int

    fun selectByPrimaryKey(id: Int?): OsGoods

    fun updateByPrimaryKeySelective(record: OsGoods): Int

    fun updateByPrimaryKey(record: OsGoods): Int

    fun selectGoodsByShopId(@Param("userId") userId: Int, @Param("shopId") shopId: Int): List<OsGoods>

}